File: JAVASCRIPT/EASY_JS_LEETCODE/goat_latin.js
/**
 * @param {string} sentence
 * @return {string}
 */
var toGoatLatin = function(sentence) {
        let words = sentence.split(' ');
        let vowels = 'aeiouAEIOU';
        for (let i = 0; i < words.length; i++) {
            if (vowels.includes(words[i][0])) {
                words[i] = words[i] + 'ma';
            } else {
                let firstLetter = words[i][0];
                words[i] = words[i].slice(1) + firstLetter + 'ma';
            }
            words[i] += 'a'.repeat(i + 1);
        }
        return words.join(' ');
};

File: PYTHON/EASY_PY_LEETCODE/goat_latin.py
from typing import *
class Solution:
    def toGoatLatin(self, sentence: str) -> str:
        a = sentence.split(' ')
        v = 'aeiouAEIOU'
        for i in range(len(a)):
            if a[i][0] in v:
                a[i] = a[i] + 'ma'
            else:
                fl = a[i][:1]
                rs = a[i][1:] + fl + 'ma'
                a[i] = rs
            a[i] += (i+1) * 'a'
        return " ".join(a)
